src/core/sock/sockinfo_tcp.cpp

```diff
@@ -56,6 +56,9 @@
 #include "sockinfo_tcp.h"
 #include "tcp_seg_pool.h"
 
+#define UNLOCK_RET(_ret)                                                                           \
```

Comment:
I'd prefer construction like this:
```C
if (condition) {
    goto error_unlock;
}
...
error_unlock:
    unlock_tcp_con();
    return -1;
```

Also, current lock/unlock_tcp_con() implementations don't do any additional logic. So, it's possible to use lock_guard. However, this approach will prohibit modification of lock/unlock_tcp_con().

Reply:
i tried to do that initially, but its not possible in this function because of compiling issues such declaring new variables in the middle of the function.